radosnamespace: add new CRD (#9733)
rook/rook: Documentation/ceph-pool-radosnamespace.md

> ---

Reviewer: How about if we don't backport this feature to 1.8 and wait for 1.9? In that case, please add to the feature list in PendingReleaseNotes.md.

Author: Yes this is fine i will add this to pendingreleasenotes

> title: RadosNamespace CRD

Author: I followed the pattern followed when we added subvolumegroup CRD, we have a new document for subvolumegroups https://rook.io/docs/rook/v1.8/ceph-fs-subvolumegroup.html, if we expand the CRD and add more details to radosnamespace CRD or the pool CRD the file content might grow.  if required I can change this and add it to pool CRD section,

> title: RadosNamespace CRD

Reviewer: Seems like the subvolumegroup doc could likewise be merged with the filesystem CRD doc. @leseb wdyt?

> title: RadosNamespace CRD

Reviewer: ?

weight: 3610
indent: true
---

{% include_relative branch.liquid %}

This guide assumes you have created a Rook cluster as explained in the main [Quickstart guide](quickstart.md)

> # CephBlockPoolRadosNamespace CRD

Reviewer: How would a volume be provisioned inside the namespace? Does the storage class have some property? Or is csi integration not implemented yet?

Rook allows creation of Ceph BlockPool
[RadosNamespaces](https://docs.ceph.com/en/latest/man/8/rbd/) through the
custom resource definitions (CRDs). BlockPool Rados Namespace is an abstraction

> for a block pool. For more information about BlockPool and namespace refer to

Reviewer: ```suggestion
for a Ceph RADOS namespace. For more information about BlockPool and namespace refer to
```

Author: It's an abstraction for a block pool. we can have `N` number of rados namespaces in a pool.

Author: sounds good :+1:

> the [Ceph docs](https://docs.ceph.com/en/latest/man/8/rbd/).

Reviewer: Could you add a sentence that gives an idea why a namespace would be useful or when it should be used?

Author: sure will do it.

## Creating daemon

To get you started, here is a simple example of a CRD to create a CephBlockPoolRadosNamespace on the CephBlockPool "replicapool".

```yaml
apiVersion: ceph.rook.io/v1
kind: CephBlockPoolRadosNamespace
metadata:

>   name: namespace-a

Author: no not required, this is the same as the concept of subvolumegroups in a filesystem. For now, this should be fine?

  namespace: rook-ceph # namespace:cluster
spec:
  # blockpoolName is the name of Ceph BlockPool. Typically it's the name of the CephBlockPool CR.

>   blockpoolName: replicapool

Reviewer 2: ```suggestion
  blockPoolName: replicapool
```

---

> title: RadosNamespace CRD

Reviewer 3: ?

```suggestion
title: CephBlockPoolRadosNamespace CRD
```

restrict access by pool).  Overloading pools for both purposes makes it hard to
do multi-tenancy because it not a good idea to have a very large number of
pools.

A namespace would be a division of a pool into separate logical namespaces. For
more information about BlockPool and namespace refer to the [Ceph
docs](https://docs.ceph.com/en/latest/man/8/rbd/)

Having multiple namespaces in a pool would allow multiple Kubernetes clusters
to share one unique ceph cluster without creating a pool per kubernetes cluster
and it will also allow to have tenant isolation between multiple tenenats in a
single kubernetes cluste without creating pool's per tenants.

Rook allows creation of Ceph BlockPool
[RadosNamespaces](https://docs.ceph.com/en/latest/man/8/rbd/) through the
custom resource definitions (CRDs).

## Creating CephBlockPoolRadosNamespace CR

To get you started, here is a simple example of a CR to create a CephBlockPoolRadosNamespace on the CephBlockPool "replicapool".

```yaml
apiVersion: ceph.rook.io/v1
kind: CephBlockPoolRadosNamespace
metadata:
  name: namespace-a
  namespace: rook-ceph # namespace:cluster
spec:

>   # blockPoolName is the name of Ceph BlockPool. Typically it's the name of the CephBlockPool CR.

Author: Yes its the blockpool CR name
